src/main/java/com/example/lesson4/EmailFile.java

```diff
@@ -0,0 +1,121 @@
+package com.example.lesson4;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.stereotype.Repository;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Scanner;
+import java.util.regex.Pattern;
+
+
+@Repository
+public class EmailFile {
```

Comment:
This component is doing too many things. Try to split the logic into multiple components and tie them later using Spring Dependency Injection. For example, you can leave all functionality related to email storage here, but validation and interaction with console should be inside separate components